backend/api/views.py

```diff
@@ -107,32 +116,105 @@ def create_job(self, request):
             raise exceptions.NotFound("Selected printer does not exist")
         job = self._create_printing_job(printer_with_perms=printer_with_perms, **serializer.validated_data)
         return Response(self.get_serializer(job).data)
+    
+    @action(detail=True, methods=['post'], name='Change job properties')
+    def change_properties(self, request, pk=None):
+        job = self.get_object()
+        serializer = CreatePrintJobRequestSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self._validate_properties(serializer.validated_data['printer'], PrintingProperties(
+            color=serializer.validated_data['color'],
+            copies=serializer.validated_data['copies'],
+            two_sides=serializer.validated_data['two_sides'],
+            pages_to_print=serializer.validated_data['pages_to_print'],
+            fit_to_page=serializer.validated_data['fit_to_page'],
+        ))
+        printer_with_perms = Printer.get_printer_for_user(user=self.request.user,
+                                                          printer_id=serializer.validated_data['printer'])
+        job = self._change_properties(printer_with_perms=printer_with_perms, **serializer.validated_data)
+        return Response(self.get_serializer(job).data)
+    
+    @action(detail=True, methods=['get'], name='Validate job properties')
+    def validate_properties(self, request, pk=None):
+        job = self.get_object()
+        printer = job.printer
+        properties = job.properties
+        self._validate_properties(printer.id, properties)
+        return Response(self.get_serializer(job).data)
+
+    @action(detail=True, methods=['get'], name='List artefacts')
+    def artefacts(self, request, pk=None):
+        job = self.get_object()
+        artefacts = job.artefacts.all().order_by('document_number')
+        serializer = JobArtefactSerializer(artefacts, many=True, context={'request': request})
+        return Response(serializer.data)
 
     def _create_printing_job(self, printer_with_perms,
                              copies: int, pages_to_print: str,
                              color: bool, two_sides: str, fit_to_page: bool, **_):
         job = GutenbergJob.objects.create(name='webrequest', job_type=JobType.PRINT, status=JobStatus.INCOMING,
                                           owner=self.request.user, printer=printer_with_perms)
-        color = color if printer_with_perms.color_allowed else False
-        two_sides = two_sides if printer_with_perms.duplex_supported else TwoSidedPrinting.ONE_SIDED
```

Comment:
Deleting these is right 👍, please remember to remove the commented out lines before marking as ready for review